### ISSUE TITLE
Creating Customer without password is directly confirmed

### ISSUE THREAD

**Issue by victortodoran** [magento/magento2#14492]
**Preconditions** 
Magento 2.*

**Steps to reproduce**
 - Create customer programatically without password using AccountManagementInterface 
 - Set Customer Account Confirmation Required in admin.
**EDIT**
 - **This applies to default Magento behavior as well**
 - Place order as guest, account will be created and customer will show as confirmed even though confirmation is required setting is set.

**Expected result**
- Newly created customer account should require confirmation

**Actual result**
- Newly created customer account is directly confirmed.
- The customer never logged in. Never confirmed the account. Yet the account is **directly confirmed**

This is deeply enforced in the core, see excerpt from ResourceModel/Customer::_beforeSave()
```
 // set confirmation key logic
        if ($customer->getForceConfirmed() || $customer->getPasswordHash() == '') {
            $customer->setConfirmation(null);
        }
```

Why would you do this? It makes no logic business wise. Why would you enforce this so deep in the core? Please see the **edit**. Please ask for some more information if needed.

**Comment by ghost** [comment]
HI @victortodoran,  thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @ansari-krish. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by ansari-ziyaurrahman** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ansari-krish. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ansari-krish, here is your [Magento instance](https://i-14492-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-14492-2-3-develop.instances.magento-community.engineering/admin](https://i-14492-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @victortodoran. Thank you for your report. 
The issue has been fixed in magento/magento2#21394 by @novikor in 2.3-develop branch
Related commit(s):
 - [4e11099e6163b5be250725c4727f85c0bba4847b](https://github.com/magento/magento2/commit/4e11099e6163b5be250725c4727f85c0bba4847b)
 - [1e55b13126e87f3b9e120743851ec7f0084eaaea](https://github.com/magento/magento2/commit/1e55b13126e87f3b9e120743851ec7f0084eaaea)
 - [1ff6e0f74d827c8e4d614fb307854d3ee307af9e](https://github.com/magento/magento2/commit/1ff6e0f74d827c8e4d614fb307854d3ee307af9e)

The fix will be available with the upcoming 2.3.3 release.
